app/controllers/vm_common.rb

```diff
@@ -1051,6 +1055,26 @@ def scan_histories
     end
   end
 
+  def parent_folder_id(vm)
+    if vm.orphaned
+      "xx-orph"
+    elsif vm.archived
+      "xx-arch"
+    elsif vm.cloud && vm.template
+      TreeBuilder.build_node_cid(vm.ext_management_system)
```

Comment:
Could this introduce an N+1?  Same question goes for vm.parent_blue_folder and vm.availability_zone below.  @kbrock?